app/data_model/submitted_responses.py

```diff
@@ -1,37 +1,44 @@
 from datetime import datetime
 from botocore.exceptions import BotoCoreError, ConnectionError as BotoConnectionError
+from flask import current_app
 from structlog import get_logger
 
+from app.globals import get_dynamodb
+
+
 logger = get_logger()
 
 
-class SubmittedResponses(object):
+def put_item(item):
+    """Insert an item to table"""
+    if item and item['valid_until']:
+        item['valid_until'] = int(item['valid_until'].strftime('%s'))  # convert to epoch
+
+    try:
+        response = _get_table().put_item(Item=item)['ResponseMetadata']['HTTPStatusCode']
+        return response == 200
+
+    except (BotoCoreError, BotoConnectionError, ConnectionError):
+        logger.warn('Could not load data into DynamoDB')
+        return False
 
-    def __init__(self, table):
-        self.table = table
 
-    def put_item(self, item):
-        """Insert an item to table"""
-        if item and item['valid_until']:
-            item['valid_until'] = int(item['valid_until'].strftime('%s'))  # convert to epoch
+def get_item(tx_id):
+    """ Get an item given its key """
+    try:
+        response = _get_table().get_item(Key={'tx_id': tx_id})
+        item = response.get('Item', None)
+        if item:
+            item['valid_until'] = datetime.utcfromtimestamp(item['valid_until'])
+        return item
 
-        try:
-            response = self.table.put_item(Item=item)['ResponseMetadata']['HTTPStatusCode']
-            return response == 200
+    except (BotoCoreError, BotoConnectionError, ConnectionError):
+        logger.warn('Could not get data from DynamoDB')
+        return None
 
-        except (BotoCoreError, BotoConnectionError, ConnectionError):
-            logger.warn('Could not load data into DynamoDB')
-            return False
 
-    def get_item(self, tx_id):
-        """ Get an item given its key """
-        try:
-            response = self.table.get_item(Key={'tx_id': tx_id})
-            item = response.get('Item', None)
-            if item:
-                item['valid_until'] = datetime.utcfromtimestamp(item['valid_until'])
-            return item
+def _get_table():
+    dynamodb = get_dynamodb()
+    table_name = current_app.config['EQ_SUBMITTED_RESPONSES_TABLE_NAME']
 
-        except (BotoCoreError, BotoConnectionError, ConnectionError):
-            logger.warn('Could not get data from DynamoDB')
-            return None
+    return dynamodb.Table(table_name)
```

Comment:
Wouldn't this fail if get_dynamodb() returns None?
See: https://github.com/ONSdigital/eq-survey-runner/pull/1486/files#diff-a160493e40efb08443761eff2a589ea2R72
Also EQ_SUBMITTED_RESPONSES_TABLE_NAME may not be set if dynamodb is not enabled.

Reply:
Good catch!